external/loaders/loaders/batches/_batch.py

```diff
@@ -147,12 +157,19 @@ def batches_from_mapper(
             add_wind_rotation_info(res),
         ]
 
-    transforms += [add_derived_data(variable_names)]
+    transforms.append(add_derived_data(variable_names))
+
+    if unstacked_dims is not None:
+        transforms.append(curry(stack)(unstacked_dims))
+        transforms.append(shuffle)
+
+    if drop_nans:
+        transforms.append(dropna)
```

Comment:
`dropna` assumes the input is stacked with a `sample` dimension, can you put this check within the block above? 

Although that creates some ambiguity in the config since dropna needs a `dim` arg and only makes sense if `unstacked_dims` is provided. Maybe we should go with your original plan of removing `dropna` from the data loading, and assume that our training data should have non nan values. I'd then worry about the case where a bunch of training data gets downloaded only for the training to start and choke on a NaN.

Reply:
As discussed, let's go with raising an exception if data is unstacked but `drop_nans` is True.

Reply:
Done.